src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -1394,7 +1450,9 @@ def _process_decln(self, parent, decl, default_visibility,
                             sym_name, str(sym.interface)))
                 sym.interface = interface
 
-    def process_declarations(self, parent, nodes, arg_list):
+    def process_declarations(self, parent, nodes, arg_list,
+                             default_visibility=Symbol.DEFAULT_VISIBILITY,
```

Comment:
This would still fail if the caller is not aware that the accessibility needs to be processed before this function and the declarations do contain some accessibility specifications.

Could we make `default_visibility=None` and then start the function with:
```
if default_visibility is None:
    (default_visibility, visibility_map) = self._parse_access_statements(node)
```

Reply:
Done plus extra test.